libraries/Network/src/NetworkManager.cpp

```diff
@@ -82,8 +45,9 @@ static esp_err_t wifi_gethostbyname_tcpip_ctx(void *param)
  */
 int NetworkManager::hostByName(const char* aHostname, IPAddress& aResult, bool preferV6)
```

Comment:
looks like `preferV6` is no longer needed

Reply:
I'm not really sure what preferV6 was supposed to be for. It seemed to be a way to say in advance what type of address you want, except the whole point of DNS is to only know the name and not the address.

i.e. www.google.com -- yesterday it was IPv4 only, today it is dual stack, tomorrow it will be IPv6 only. But the code shouldn't care.

The programmer won't know where the device/code will be used, so can't specify what to use in advance.

The rules for destination addresses of RFC 6724 are to use whatever matches, i.e. if you have a public IPv6 and the destination is public IPv6, then use that; otherwise if you only have IPv4 then use that. This needs to be determined dynamically at run time. (what the patch to ESP-LWIP does, or the work around does as a short cut).

Reply:
Note that in some utility programs you might want to display all DNS lookups. Unfortunately LWIP is limited to only one result, although you could get one IPv6 and one IPv4.

Although in that case you probably want at least 3 options -- the default to give whatever matches (4 or 6), and then an explicit way to look up either v4 or v6.

Note that a utility program might be running in an IPv4-only network, so the default will give a v4 address, but they might want to show the user both available addresses.

Although at that point, for such a utility, probably Arduino is not the right solution (i.e. go to underlying ESP-IDF).

I will remove the option (I think it was only added as part of the original refactoring).

Reply:
it was there exactly so some that want to prefer using V6 address, can say so. Then the DNS will return the addresses in the requested order. By default it will return the v4 if available.

Reply:
Okay, but I never understood how a programmer would know in advance what type of address is going to be needed.

Dynamic determination is much better, because that allows a device, or sample code, to run in any network.

Now that I have that working, there is no need for a programmer to have to use the option (and change configuration/recompile every time they change networks). The dynamic code will work in any network, for any destination.

Reply:
yes, I am fine with that. Please remove the argument :)